cli/compose/convert/service.go

```diff
@@ -214,18 +214,21 @@ func convertServiceNetworks(
 		if !ok && networkName != defaultNetwork {
 			return nil, errors.Errorf("undefined network %q", networkName)
 		}
-		var aliases []string
-		if network != nil {
-			aliases = network.Aliases
-		}
 		target := namespace.Scope(networkName)
 		if networkConfig.External.External {
 			target = networkConfig.External.Name
 		}
-		nets = append(nets, swarm.NetworkAttachmentConfig{
-			Target:  target,
-			Aliases: append(aliases, name),
-		})
+		netAttachConfig := swarm.NetworkAttachmentConfig{
+			Target: target,
+		}
+		if container.NetworkMode(target).IsUserDefined() {
```

Comment:
@dnephin i dont understand your question. But if we try to configure network alias in host / bridge network, it will fail clearly stating network alias is not supported.

Since the alias is done automatically in stack, we can never support `--network=host` in stack unless we remove this alias. 

am not sure if that answers your question. But yes, pls give it a try.